include/nighthawk/adaptive_load/step_controller.h

```diff
@@ -66,7 +66,7 @@ using StepControllerPtr = std::unique_ptr<StepController>;
  * A factory that must be implemented for each StepController plugin. It instantiates the
  * specific StepController class after unpacking the plugin-specific config proto.
  */
-class StepControllerConfigFactory : public Envoy::Config::TypedFactory, public ConfigValidator {
+class StepControllerConfigFactory : public virtual Envoy::Config::TypedFactory, public virtual ConfigValidator {
```

Comment:
Can we avoid multiple implementation inheritance here with some refactoring or changes to is-a vs. has-a relationships? Google/Envoy C++ style discourages multiple implementation inheritance.

Reply:
Yes, I am removing the shared concrete NullConfigValidator that we were going to inherit from. Instead a plugin factory that doesn't need to validate anything can just explicitly override ValidateConfig() with a body that immediately returns absl::OkStatus(). No other ConfigValidator implementations will need to be shared because they will all be plugin-specific.

Reply:
For now, I'm eliminating the multiple inheritance but not shaking up plugin_loader or all the tests written according to these interfaces. Filed #465 to consider whether a refactor will clean it up further, after all these PRs land.